### PR TITLE
Porting into latest version of transformers & torch

### PR DESCRIPTION
**Pull request by woosukji**
Hi,

Recently I've been working on Conditional Text Generation for [2021 OSAM Hackathon](https://github.com/osamhack2021?q=pplm&type=&language=&sort=) in Korea,
[Our team](https://github.com/osamhack2021/AI_WEB_POOL_YD) tried your paper code, in order to generate some sentences suitable for job application resume,
and during the process we had to do some modification to match our dependencies.

Here I'd like to share the outcomes, with support to latest version of `torch` and `transformers`,
although this version only takes care of PPLM-BOW (i.e. irrelevant dependencies like `torchtext` might cause some errors when trying out PPLM-DISCRIM or so).
BOW scenario was fully tested and seems to be working well.

The main consideration was the discrepancy of input and output schema of `transformers.GPT2LMHeadModel`.
With older version, `past` values would encapsulate query & key tensors (of attention layers) in the same large tensor,
but now they're seperated into tuples, demanding a deeper method for adding two past tensors.

Anyway, just for someone who'd be interested in bringing your awesome works into current projects.

thx!

### CODE REVIEW

**Review by w4nderlust — APPROVED** [review]
they basically added an additinal dimension i believe and you adapted to make it work with that in multiple places, do i read it correctly. All looks food to me

**Comment by w4nderlust** [comment]
@dathath have you tested it out with our usual prompts by any chance? just to make sure generated texts are comaprable

**Comment by akshaylive** [comment]
`requirements.txt` should be updated as well, but would love to see some traction on this :)

File: run_pplm.py [hunk 1]
```diff
@@ -375,7 +402,6 @@ def get_bag_of_words_indices(bag_of_words_ids_or_paths: List[str], tokenizer) ->
             words = f.read().strip().split("\n")
         bow_indices.append(
             [tokenizer.encode(word.strip(),
-                              add_prefix_space=True,
```
Comment (akshaylive) [inline run_pplm.py:378]: This line is required. Each line in BOW needs a prefix, without which, `bow_indices` is correct. This drastically changes the output of BOW model output due to incorrect gradient computation.

**Review by akshaylive — CHANGES_REQUESTED** [review]
Minor change needed to make BOW model work.

File: run_pplm.py (hunk 1 above)
Comment (akshaylive) [reply run_pplm.py:378]: Additionally, new L241 needs
```
output_hidden_states=True,
return_dict=False
```

**Pull request closed without merging**